src/react/components/common/tagsInput/tagEditorModal/tagEditorModal.test.tsx

```diff
@@ -0,0 +1,121 @@
+import { mount } from "enzyme";
+import React from "react";
+import MockFactory from "../../../../../common/mockFactory";
+import TagEditorModal, { ITagEditorModalProps } from "./tagEditorModal";
+// tslint:disable-next-line:no-var-requires
+const TagColors = require("../tagColors.json");
+
+describe("Tag Editor Modal", () => {
+
+    function createComponent(props: ITagEditorModalProps) {
+        return mount(
+            <TagEditorModal {...props} />,
+        );
+    }
+
+    const tag = MockFactory.createTestTag();
+
+    it("modal is initialized properly", () => {
+        const onCancel = jest.fn();
+        const onOk = jest.fn();
+        const wrapper = createComponent({
+            tag,
+            showModal: false,
+            onCancel,
+            onOk,
+        });
```

Comment:
Is it possible to do these few lines in a beforeEach to reduce code redundancy in the tests below?

Reply:
So that's what I had initially, but I was hitting some race conditions with some assertions. For example, the `onOk` handler is called with different tags in different tests. But I can probably move the `onCancel` and a default `wrapper` to the beforeEach :)